gokey.go

```diff
@@ -11,25 +11,36 @@ import (
 	"io"
 )
 
-func GetPass(password, realm string, seed []byte, spec *PasswordSpec) (string, error) {
+func getReader(password, realm string, seed []byte, allowUnsafe bool) (io.Reader, error) {
 	var rng io.Reader
 	var err error
 
-	if seed == nil {
-		rng = NewDRNG(password, realm)
-	} else {
+	if seed != nil {
 		rng, err = NewDRNGwithSeed(password, realm, seed)
 		if err != nil {
-			return "", err
+			return nil, err
 		}
+	} else if allowUnsafe {
+		rng = NewDRNG(password, realm)
+	} else {
+		return nil, errors.New("generating keys without strong seed is not allowed")
+	}
+
+	return rng, nil
+}
+
+func GetPass(password, realm string, seed []byte, spec *PasswordSpec) (string, error) {
+	rng, err := getReader(password, realm+"-pass", seed, true)
+	if err != nil {
+		return "", err
 	}
 
 	gen := &KeyGen{rng}
 	return gen.GeneratePassword(spec)
 }
 
 func GetKey(password, realm string, seed []byte, keyType int, allowUnsafe bool) (crypto.PrivateKey, error) {
-	rng, err := GetRaw(password, realm, seed, allowUnsafe)
+	rng, err := getReader(password, realm+fmt.Sprintf("-key(%v)", keyType), seed, allowUnsafe)
```

Comment:
I think we should try to map back to a string key type rather than to just int number here. What if in the future we might want to have alternative implementations in other languages, which may internally interpret key type enum differently?

Maybe somehow reuse https://github.com/cloudflare/gokey/blob/master/cmd/gokey/main.go#L31-L36 or put it lower in the stack, so this function can access defined string literals.